File: multilibrary/src/main/java/com/zhongjh/albumcamerarecorder/settings/RecorderSetting.java
package com.zhongjh.albumcamerarecorder.settings;

import com.zhongjh.albumcamerarecorder.settings.api.RecorderSettingApi;

/**
 * 录音机
 */
public final class RecorderSetting implements RecorderSettingApi {

    private final RecordeSpec mRecordeSpec;

    public RecorderSetting() {
        mRecordeSpec = RecordeSpec.getCleanInstance();
    }

    @Override
    public RecorderSetting duration(int duration) {
        mRecordeSpec.duration = duration;
        return this;
    }

    @Override
    public RecorderSetting minDuration(int minDuration) {
        mRecordeSpec.minDuration = minDuration;
        return this;
    }

}

File: multilibrary/src/main/java/com/zhongjh/albumcamerarecorder/settings/RecorderSetting.kt
package com.zhongjh.albumcamerarecorder.settings

import com.zhongjh.albumcamerarecorder.settings.api.RecorderSettingApi

/**
 * 录音机
 * @author zhongjh
 */
class RecorderSetting : RecorderSettingApi {

    private val mRecordeSpec: RecordeSpec = RecordeSpec.cleanInstance

    override fun duration(duration: Int): RecorderSetting {
        mRecordeSpec.duration = duration
        return this
    }

    override fun minDuration(minDuration: Int): RecorderSetting {
        mRecordeSpec.minDuration = minDuration
        return this
    }

}